validator/keymanager/remote-web3signer/keymanager.go

```diff
@@ -113,166 +113,283 @@ func getSignRequestJson(ctx context.Context, validator *validator.Validate, requ
+	if err != nil {
+		return nil, err
+	}
+	if err = validator.StructCtx(ctx, voluntaryExitRequest); err != nil {
+		return nil, err
+	}
+	voluntaryExitSignRequestsTotal.Inc()
+	return json.Marshal(voluntaryExitRequest)
+}
+
+func handleSyncMessageBlockRoot(ctx context.Context, validator *validator.Validate, request *validatorpb.SignRequest, genesisValidatorsRoot []byte) ([]byte, error) {
+	syncCommitteeMessageRequest, err := web3signerv1.GetSyncCommitteeMessageSignRequest(request, genesisValidatorsRoot)
+	if err != nil {
+		return nil, err
+	}
+	if err = validator.StructCtx(ctx, syncCommitteeMessageRequest); err != nil {
+		return nil, err
+	}
+	syncCommitteeMessageSignRequestsTotal.Inc()
+	return json.Marshal(syncCommitteeMessageRequest)
+}
+
+func handleSyncAggregatorSelectionData(ctx context.Context, validator *validator.Validate, request *validatorpb.SignRequest, genesisValidatorsRoot []byte) ([]byte, error) {
+	syncCommitteeSelectionProofRequest, err := web3signerv1.GetSyncCommitteeSelectionProofSignRequest(request, genesisValidatorsRoot)
+	if err != nil {
+		return nil, err
+	}
+	if err = validator.StructCtx(ctx, syncCommitteeSelectionProofRequest); err != nil {
+		return nil, err
+	}
+	syncCommitteeSelectionProofSignRequestsTotal.Inc()
+	return json.Marshal(syncCommitteeSelectionProofRequest)
+}
+
+func handleContributionAndProof(ctx context.Context, validator *validator.Validate, request *validatorpb.SignRequest, genesisValidatorsRoot []byte) ([]byte, error) {
+	contributionAndProofRequest, err := web3signerv1.GetSyncCommitteeContributionAndProofSignRequest(request, genesisValidatorsRoot)
+	if err != nil {
+		return nil, err
+	}
+	if err = validator.StructCtx(ctx, contributionAndProofRequest); err != nil {
+		return nil, err
+	}
+	syncCommitteeContributionAndProofSignRequestsTotal.Inc()
+	return json.Marshal(contributionAndProofRequest)
+}
+
+func handleRegistration(ctx context.Context, validator *validator.Validate, request *validatorpb.SignRequest, genesisValidatorsRoot []byte) ([]byte, error) {
+	validatorRegistrationRequest, err := web3signerv1.GetValidatorRegistrationSignRequest(request)
+	if err != nil {
+		return nil, err
+	}
+	if err = validator.StructCtx(ctx, validatorRegistrationRequest); err != nil {
+		return nil, err
+	}
+	validatorRegistrationSignRequestsTotal.Inc()
+	return json.Marshal(validatorRegistrationRequest)
+}
+
+func handleBlob(ctx context.Context, validator *validator.Validate, request *validatorpb.SignRequest, genesisValidatorsRoot []byte) ([]byte, error) {
+	blobRequest, err := web3signerv1.GetBlobSignRequest(request)
+	if err != nil {
+		return nil, err
+	}
+	if err = validator.StructCtx(ctx, blobRequest); err != nil {
+		return nil, err
+	}
+	blobSignRequestsTotal.Inc()
+	return json.Marshal(blobRequest)
+}
+
+func handleBlindedBlob(ctx context.Context, validator *validator.Validate, request *validatorpb.SignRequest, genesisValidatorsRoot []byte) ([]byte, error) {
```

Comment:
genesisValidatorsRoot is not used, is this to conform to some interface? Several methods in this file that have unused params.

Reply:
this is a good point I can probably remove, i had to make this function because of the linter cognit I can remove for many of these.

Reply:
oof ok it actually needs the fork info so thanks for catching